src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -169,58 +172,60 @@ const usePushNotificationHook = () => {
-            updatedAt: message.created_at,
-            rawJson: {},
-            attachmentJson: message.attachment,
-            user: message.user,
-            status: 'sent',
-            isMe: false,
-            isContinuous: false
-          });
-          await chatSchema.save(localDb);
-        } else {
-          const chatSchema = new ChatSchema({
-            id: remoteMessage.data?.messages_id,
-            channelId: remoteMessage.data?.channel_id,
-            userId: remoteMessage.data?.user_id,
-            message: remoteMessage.data?.message,
-            type: remoteMessage.data?.type,
-            createdAt: remoteMessage.data?.created_at,
-            updatedAt: remoteMessage.data?.created_at,
-            rawJson: {},
-            attachmentJson: remoteMessage.data?.attachment,
-            user: null,
-            status: remoteMessage.data?.status,
-            isMe: false,
-            isContinuous: false
-          });
-          await chatSchema.save(localDb);
+          if (!response?.id) {
+            const channel = new ChannelList({
+              // craete channel payload
+              id: remoteMessage.data?.channel_id,
+              channelType: 'PM'
+            });
+            await fetchChannelDetail(channel); // insert channel detail
+          }
+          if (remoteMessage.data?.is_big_message === 'true') {
+            const {message: messageRes} = await getMessageDetail(
+              remoteMessage.data?.messages_id as string
+            );
+            const {message} = messageRes;
+            const chatSchema = new ChatSchema({
+              id: message.id,
+              channelId: message.channel.id,
+              userId: message.user.id,
+              message: message.text,
+              type: message.type,
+              createdAt: message.created_at,
+              updatedAt: message.created_at,
+              rawJson: '{}',
+              attachmentJson: message.attachments,
+              user: message.user,
+              status: 'sent',
+              isMe: false,
+              isContinuous: false
+            });
+            await chatSchema.save(localDb);
+          } else {
+            const chatSchema = new ChatSchema({
+              id: remoteMessage.data?.messages_id,
+              channelId: remoteMessage.data?.channel_id,
+              userId: remoteMessage.data?.user_id,
+              message: remoteMessage.data?.message,
+              type: remoteMessage.data?.type,
+              createdAt: remoteMessage.data?.created_at,
+              updatedAt: remoteMessage.data?.created_at,
+              rawJson: '{}',
+              attachmentJson: remoteMessage.data?.attachments,
+              user: null,
+              status: remoteMessage.data?.status,
+              isMe: false,
+              isContinuous: false
+            });
+            await chatSchema.save(localDb);
+          }
+          setIsLoadingFetchAllMessage(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code has been significantly modified to handle incoming push notifications for new messages. The changes include setting a loading state, fetching channel details if they don't exist, and saving the new message to the local database.

However, there are a few potential issues:

1. There's no error handling for the various async operations (fetching channel details, getting message details, saving to the local database). If any of these operations fail, the app could end up in an inconsistent state. Consider wrapping these operations in try-catch blocks and handling errors appropriately.

2. The loading state (`setIsLoadingFetchAllMessage`) is set to true at the start of the block but only set to false at the very end. If any operation fails and throws an error before reaching the end, the loading state will remain true indefinitely. To fix this, you could use a finally block to ensure the loading state is always set to false after these operations.

3. The creation of the `ChatSchema` objects is quite verbose and duplicated for the two cases (`is_big_message === 'true'` and else). Consider creating a helper function to construct these objects to reduce code duplication.

4. The `rawJson` field is being set to `'{}'`, which is a string, not an object. If this field is expected to be an object elsewhere in your code, this could cause issues. Make sure this is the intended behavior.

Here's a simplified example of how you might address these issues:

```diff
+ try {
    setIsLoadingFetchAllMessage(true);
    const response = await ChannelList.getById(
      localDb,
      remoteMessage.data?.channel_id as string
    );
    if (!response?.id) {
      const channel = new ChannelList({
        id: remoteMessage.data?.channel_id,
        channelType: 'PM'
      });
      await fetchChannelDetail(channel);
    }
    let chatSchema;
    if (remoteMessage.data?.is_big_message === 'true') {
      const {message: messageRes} = await getMessageDetail(
        remoteMessage.data?.messages_id as string
      );
      const {message} = messageRes;
      chatSchema = createChatSchemaFromMessage(message);
    } else {
      chatSchema = createChatSchemaFromRemoteData(remoteMessage.data);
    }
    await chatSchema.save(localDb);
+ } catch (error) {
+   // handle error
+ } finally {
    setIsLoadingFetchAllMessage(false);
+ }
```
In this example, `createChatSchemaFromMessage` and `createChatSchemaFromRemoteData` are hypothetical helper functions that you would need to implement. They should take a message or remote data object, respectively, and return a new `ChatSchema` object.


<!-- This is an auto-generated comment by OSS CodeRabbit -->